drivers/cps-hid.c

```diff
@@ -262,6 +262,79 @@ static int cps_claim(HIDDevice_t *hd) {
 	}
 }
 
+/* CPS Models like CP900EPFCLCD return a syntactically legal but incorrect
+ * Report Descriptor whereby the Input High Transfer Max/Min values are used
+ * for the Output Voltage Usage Item limits. This corrects them by finding and 
+ * applying the Input Voltage limits as being more appropriate. 
+ */
+
+HIDData_t *FindReport(HIDDesc_t *pDesc, uint8_t ReportID, HIDNode_t node)
+{
+	size_t	i;
+
+	for (i = 0; i < pDesc->nitems; i++) {
+		HIDData_t *pData = &pDesc->item[i];
+
+		if (pData->ReportID != ReportID) {
+			continue;
+		}
+
+		HIDPath_t * pPath = &pData->Path;
+		uint8_t size = pPath->Size;
+		if (size == 0 || pPath->Node[size-1] != node) {
+			continue;
+		}
+
+		return pData;
+	}
+
+	return NULL;
+}
+
+static int cps_fix_report_desc(HIDDevice_t *pDev, HIDDesc_t *pDesc) {
+	HIDData_t *pData;
+
+	int vendorID = pDev->VendorID;
+	int productID = pDev->ProductID;
+	if (vendorID != CPS_VENDORID || productID != 0x0501) {
+		return 0;
+	}
+
+	upsdebugx(3, "Attempting Report Descriptor fix for UPS: Vendor:%04x, Product:%04x", vendorID, productID);
+
+	/* Apply the fix cautiously by looking for input voltage, high voltage transfer and output voltage report usages.
+	 * If the output voltage log min/max equals high voltage transfer log min/max then the bug is present.
+	 * To fix it copy the input voltage log min/max settings as reasonable values for the output voltage limits.
+	 */
+
+	if ((pData=FindReport(pDesc, 15, (PAGE_POWER_DEVICE<<16)+USAGE_VOLTAGE))) {
```

Comment:
I my opinion we should skip report 15 (0xe) and change output voltage:
- LogMin to 0
- LogMax leave as it is or even increase it (current value 270)

According to the [specification ](https://www.cyberpower.com/eu/pl/File/GetFileSampleByType?fileId=DS-18080023-01&fileType=Download%20Center&fileSubType=FileOriginal) for EU version of the UPS "Input Voltage Range" is "170 ~ 270". 
which means that the input voltage up to 270 V can be accepted, before system will switch to a battery.
If UPS will switch to battery I'm not sure if it will report input voltage higher than 270V, I don't have equipment to verify it.